HealthcareSystem/GraphicalEditor/MainWindow.xaml.cs

```diff
@@ -29,12 +33,15 @@ namespace GraphicalEditor
     public partial class MainWindow : Window, INotifyPropertyChanged
     {
         private Canvas _canvas;
-
+        private MapObjectController mapObjectController;
+        private IMapObjectServices mapObjectServices;
+        private IMapObjectRepository mapObjectRepository;
```

Comment:
Which of these fields are actually needed? Why have instances of a service, a repository, and a controller? Also, some fields have an underscore and some don't, make it consistent.

Reply:
i use all of them

Reply:
You do not use all of them. Only the controller is used, the rest are initialized in the constructor and never used. It wouldn't make sense to use all of them anyway, since the controller uses the service, and the service uses the repository. So you really only need the controller as a field.

Reply:
i need repository to create service and service to create controller

Reply:
You need them locally in the constructor then since there's no DI container, but that does not mean you need them as fields.

Reply:
Okay, l removed those properties